tests/cephfs/snapshot_clone/cephfs_snap_utils.py

```diff
@@ -0,0 +1,435 @@
+        Returns: None
+        """
+        out, rc = client.exec_command(
+            sudo=True,
+            cmd=f"ceph config set mgr mgr/snap_schedule/allow_m_granularity {allow}",
+        )
+        log.info(out)
+
+    def create_snap_schedule(self, snap_params):
+        """
+        To create snapshot schedule on ceph FS and activate it. It also verifies that schedule is set.
+        Args:
+            client: ceph client to run cmd
+            snap_params : a dict data type with below key-value pairs,
+                Required:
+                    path : a path for scheduled snapshot to be created, type - str
+                    sched : a schedule to be applied to snapshot schedule ass, type - str, for eg., 2M1h
+                    fs_name : cephfs volume name
+                Optional:
+                    start_time : a iso 8601 format time value, from when schedule is to be applied. type -str
+        Returns: 0 : success, 1 : failure
+        """
+        client = snap_params["client"]
+        sched_cmd = (
+            f"ceph fs snap-schedule add {snap_params['path']} {snap_params['sched']}"
+        )
+        if snap_params.get("start_time"):
+            sched_cmd += f" {snap_params['start_time']} "
+        sched_cmd += f" --fs {snap_params['fs_name']}"
+
+        activate_cmd = f"ceph fs snap-schedule activate {snap_params['path']} --fs {snap_params['fs_name']}"
+        for cmd in [sched_cmd, activate_cmd]:
+            out, rc = client.exec_command(sudo=True, cmd=cmd)
+            log.info(out)
+        if snap_params["validate"] is True:
+            cmd = f"ceph fs snap-schedule status {snap_params['path']} --fs {snap_params['fs_name']} -f json"
+            out, rc = client.exec_command(sudo=True, cmd=cmd)
+            sched_status = json.loads(out)
+            for sched_item in sched_status:
+                log.info(f"{sched_item}")
+                if (
+                    sched_item["path"] == snap_params["path"]
+                    and sched_item["active"] is True
+                    and sched_item["schedule"] == snap_params["sched"]
+                ):
+                    log.info("Snap schedule is verified.")
+                    return 0
+            log.error(f"Snap schedule verification failed : {out}")
+            return 1
+        return 0
+
+    def get_snap_sched_status(self, client, path):
+        """
+        To get snap-schedule list for path on ceph FS .
+        Args:
+            client: ceph client to run cmd
+            path : path set for snap-schedule, type - str
+        Returns: a list, each item referring to a specfic snap-schedule on a path
+        """
+        cmd = f"ceph fs snap-schedule list {path} --recursive=true"
+        cmd_out, rc = client.exec_command(sudo=True, cmd=cmd)
+        sched_list = cmd_out.split()
+        return sched_list
+
+    def get_sched_snap_list(self, client, client_snap_path):
```

Comment:
previous func name "def get_snap_sched_status" and "def get_sched_snap_list" might be confusing.
can you rename "def get_sched_snap_list" to " def get_snaps_from_mounts" or something similar

Reply:
i have renamed it as,
get_snap_sched_status -> get_snap_schedule_list
get_sched_snap_list -> get_scheduled_snapshots